qiskit_aer/circuit/aer_circuit.py

```diff
@@ -0,0 +1,291 @@
+"""Directly simulatable circuit in Aer."""
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2017-2023.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+from typing import List, Optional
+from copy import copy
+
+from qiskit.circuit import QuantumCircuit, Clbit, ParameterExpression
+from qiskit.tools.parallel import parallel_map
+from qiskit.providers.options import Options
+from qiskit.qobj import QobjExperimentHeader
+from qiskit_aer.aererror import AerError
+# pylint: disable=import-error, no-name-in-module
+from qiskit_aer.backends.controller_wrappers import AerCircuit_
+
+
+class AerCircuit:
```

Comment:
I'm a bit uncertain about the need for this class. Couldn't we just go from `QuantumCircuit` to `AerCircuit_` directly in a for loop over `QuantumCircuit.data`. The only use for `AerCircuit` I saw was to call `AerCircuit.native_gates` which then gets passed to the execute function. I feel like we can remove a layer in the hierarchy here without too much effort. Just basically make `generate_aer_circuit` be the equivalent of `AerCircuit.native_gates` and return a  `Aercircuit_`  objects directly instead of wrapping it in a class.

Reply:
I'm thinking that `backend.run()` can take `AerCircuit` for repeatedly call the same circuit with different parameters in Aer (such as by `Estimator`). I guess that python class is better for some future utilities.
Another reason is serielization. pybind11 supports [pickling](https://pybind11.readthedocs.io/en/stable/advanced/classes.html#pickling-support) but a bit complex to handle `AER::Circuit`. Instead, if we just pickle current python `AerCircuit`, only pickling `QuantumCircuit`s (and some other fields) is necessary.